multiqc/modules/bamdst/bamdst.py

```diff
@@ -0,0 +1,419 @@
+    if cfg["exclude_contigs"]:
+        log.debug(
+            f"Configuration: would exclude these contigs, unless all would be filtered: {', '.join(cfg['exclude_contigs'])}"
+        )
+
+    cutoff = cfg.get("perchrom_fraction_cutoff", 0.0)
+    try:
+        cutoff = float(cutoff)
+    except ValueError:
+        cutoff = 0.0
+    if cutoff != 0.0:
+        log.debug(
+            f"Configuration: coverage cutoff of contigs is {cutoff * 100.0}%, unless everything would be filtered"
+        )
+    cfg["perchrom_fraction_cutoff"] = cutoff
+
+    return cfg
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="Bamdst",
+            anchor="bamdst",
+            href="https://https://github.com/shiquan/bamdst",
+            info="""is a lightweight tool to stat the depth coverage of target regions of bam file(s)""",
+            # doi="", # No DOI
+        )
+
+        self.cfg = _read_config()
+        data_by_sample = dict()
+        data_by_chromosome_by_sample = dict()
+        for f in self.find_log_files("bamdst/coverage"):
+            if data := self._parse_coverage_report(f):
+                data_by_sample[f["s_name"]] = data
+
+                # Parse "chromosomes.report" sitting next to the "coverage.report" file:
+                if os.path.exists(chroms_path := os.path.join(f["root"], "chromosomes.report")):
+                    if data := self._parse_chromosomes_report(chroms_path, f["s_name"]):
+                        data_by_chromosome_by_sample[f["s_name"]] = data
+
+        data_by_sample = self.ignore_samples(data_by_sample)
+        data_by_chromosome_by_sample = self.ignore_samples(data_by_chromosome_by_sample)
+        if len(data_by_sample) == 0:
+            raise ModuleNoSamplesFound
+
+        log.info(f"Found {len(data_by_sample)} reports")
+        self.write_data_file(data_by_sample, "multiqc_bamdst_coverage")
+        self.write_data_file(data_by_chromosome_by_sample, "multiqc_bamdst_chromosomes")
+
+        self.build_tables(data_by_sample)
+        self._build_per_chrom_plot(self._filter_contigs(data_by_chromosome_by_sample))
+
+    def _parse_coverage_report(self, f: Dict) -> Dict:
+        """
+        Parse one coverage report.
+        """
+        data = dict()
+        s_name = None
+        version = None
+        for line in f["f"].split("\n"):
+            line = line.strip()
+            if line.startswith("## Files : "):
+                # Split a line like "## Files : example/test1.bam example/test2.bam "
+                # Cannot split by spaces because the file names can contain spaces as well
+                path_line = line.split(":")[1].strip().replace(".bam", "<EXT>").replace(".cram", "<EXT>")
+                names = [self.clean_s_name(sn) for sn in path_line.split("<EXT>") if sn]
+                f["s_name"] = "-".join(names)  # concatenating file names
```

Comment:
Maybe tweak the code to not overwrite `f["s_name"]` if no valid filenames are found, so that it falls back to using the filename.

```suggestion
                if names:
                    f["s_name"] = "-".join(names)  # concatenating file names
```